QuizApp/View/Login/LoginViewController.swift

```diff
@@ -79,20 +81,24 @@ class LoginViewController: UIViewController {
         }
         loginView.errorLabel.isHidden = true
     }
+    
     @objc func buttonPressed(_ button: UIButton) {
-        let status = DataService().login(email: loginView.emailField.text!, password: loginView.passwordField.text!)
-        switch status {
-        case .success:
-            print("Email: ", loginView.emailField.text!)
-            print("Password: ", loginView.passwordField.text!)
-            router.showQuizzesViewControllerAsRoot()
-        default:
-            print(status)
-            loginView.errorLabel.text = "Email or password is incorrect!"
-            loginView.errorLabel.isHidden = false
+        LoginPresenter(router: router).login(username: loginView.emailField.text!, password: loginView.passwordField.text!) { [self]
```

Comment:
Store `presenter` in property and inject it in your init. Remove router from LoginViewController. This controller doesn't need to know about the router.